### PR TITLE
Update action bottom sheet + Fix some UI

### PR DESCRIPTION
**Pull request by LunarX**
Depends on https://github.com/Infomaniak/android-core/pull/28

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 1]
```diff
@@ -121,4 +130,16 @@ object UiUtils {
             setColor(newColor)
         }
     }
+
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:134]:
```suggestion
```

File: app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt [hunk 2]
```diff
@@ -121,4 +130,16 @@ object UiUtils {
...
+    fun Fragment.handleJunkBottomSheet() {
+        getBackNavigationResult<Bundle>(JUNK_BOTTOM_SHEET_NAV_KEY) {
+            val shouldOpenJunk = it.getBoolean(SHOULD_OPEN_JUNK)
+            val messageUid = it.getString(MESSAGE_UID)
+            val threadUid = it.getString(THREAD_UID)
+
+            val args = JunkBottomSheetDialogArgs(threadUid!!, messageUid).toBundle()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt:141]:
```suggestion
            val threadUid = it.getString(THREAD_UID)!!

            val args = JunkBottomSheetDialogArgs(threadUid, messageUid).toBundle()
```

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.models.Folder.FolderRole
+import com.infomaniak.mail.data.models.message.Message
+import com.infomaniak.mail.databinding.BottomSheetJunkBinding
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.utils.notYetImplemented
+
+class JunkBottomSheetDialog : BottomSheetDialogFragment() {
+
+    lateinit var binding: BottomSheetJunkBinding
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt:37]:
```suggestion
    private lateinit var binding: BottomSheetJunkBinding
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt [hunk 4]
```diff
@@ -22,26 +22,68 @@ import android.os.Bundle
...
-import com.infomaniak.mail.data.models.message.Message
 import com.infomaniak.mail.databinding.BottomSheetActionsMenuBinding
 import com.infomaniak.mail.ui.MainViewModel
 import com.infomaniak.mail.utils.getAttributeColor
 import com.google.android.material.R as RMaterial
 
-open class ActionsBottomSheetDialog : BottomSheetDialogFragment() {
+abstract class ActionsBottomSheetDialog : BottomSheetDialogFragment() {
 
     lateinit var binding: BottomSheetActionsMenuBinding
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt:38]:
```suggestion
    protected lateinit var binding: BottomSheetActionsMenuBinding
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    private fun setSpamUi(message: Message?) {
+        val isSpam = message?.isSpam ?: mainViewModel.isCurrentFolderRole(FolderRole.SPAM)
+        binding.spam.setText(if (isSpam) R.string.actionNonSpam else R.string.actionSpam)
+    }
+
+    private fun ActionItemView.setClosingOnClickListener(callback: (() -> Unit)) {
+        setOnClickListener {
+            callback()
+            findNavController().popBackStack()
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt:76]:
```suggestion
            findNavController().popBackStack()
            callback()
```

Calling popBackStack after the callback will close any dialog or fragment callback could navigate to. For example the future dialog for phishing confirmation

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return BottomSheetJunkBinding.inflate(inflater, container, false).also { binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) = with(binding) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val threadUid = navigationArgs.threadUid
+        val messageUid = navigationArgs.messageUid
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt:49]: Use directly navigationArgs values without creating a val

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MessageActionsBottomSheetDialog.kt [hunk 7]
```diff
@@ -19,54 +19,81 @@ package com.infomaniak.mail.ui.main.thread.actions
...
+
+                override fun onPrint() {
+                    notYetImplemented()
+                }
+
+                override fun onReportDisplayProblem() {
+                    notYetImplemented()
+                }
+
+
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MessageActionsBottomSheetDialog.kt:80]:
```suggestion

```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ThreadActionsBottomSheetDialog.kt [hunk 8]
```diff
@@ -46,35 +45,67 @@ class ThreadActionsBottomSheetDialog : ActionsBottomSheetDialog() {
...
+
+                override fun onPrint() {
+                    notYetImplemented()
+                }
+
+                override fun onReportDisplayProblem() {
+                    notYetImplemented()
+                }
+
+
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ThreadActionsBottomSheetDialog.kt:92]:
```suggestion

```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt (hunk 5 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt:76]: You can also use this with setClosingOnClickListener to navigate to JunkBottomSheet without backNavigationResults

**Review by FabianDevel — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**